CustomGlobalExceptionHandler implemented

jszymanski97 opened the pull request:
CustomGlobalExceptionHandler implemented for returning clear error messages and also returning custom exception for invalid id

CreateBookRequestDto.java - Validation annotations added

jszymanski97 committed:
CustomGlobalExceptionHandler implemented for returning clear error messages and also returning custom exception for invalid id
CreateBookRequestDto.java - Validation annotations added

slade13 commented on src/main/java/mate/project/controller/BookController.java:
```diff
@@ -1,5 +1,6 @@
 package mate.project.controller;
 
+import jakarta.validation.Valid;
```
The @Valid annotation is used to perform bean validation on the request body. Ensure that corresponding validation annotations are present in the CreateBookRequestDto class.

slade13 commented on src/main/java/mate/project/controller/BookController.java:
```diff
@@ -35,7 +36,7 @@ public BookDto getBookById(@PathVariable Long id) {
     @PostMapping
-    public BookDto createBook(@RequestBody CreateBookRequestDto bookDto) {
+    public BookDto createBook(@RequestBody @Valid CreateBookRequestDto bookDto) {
         return bookService.save(bookDto);
```
In the method createBook, the return of bookService.save might return null, which would be a violation of the method's return type. Make sure to handle a possible null value.

slade13 commented on src/main/java/mate/project/dto/CreateBookRequestDto.java:
```diff
@@ -1,11 +1,18 @@
+        @NotNull
+        @Min(0)
         BigDecimal price,
         String description,
```
The 'description' field of the CreateBookRequestDto class is not validated. Consider adding appropriate validation constraints to this field.

slade13 commented on src/main/java/mate/project/dto/CreateBookRequestDto.java:
```diff
@@ -1,11 +1,18 @@
+        @Min(0)
         BigDecimal price,
         String description,
         String coverImage
```
The 'coverImage' field of the CreateBookRequestDto class is not validated. Consider adding appropriate validation constraints to this field.

slade13 commented on src/main/java/mate/project/exception/CustomGlobalExceptionHandler.java:
```diff
@@ -0,0 +1,59 @@
+package mate.project.exception;
+
+import java.time.LocalDate;
```
Consider using java.time.ZonedDateTime or java.time.Instant instead of LocalDate in order to include the time and timezone information along with the date.

slade13 commented on src/main/java/mate/project/exception/CustomGlobalExceptionHandler.java:
```diff
@@ -0,0 +1,59 @@
+import java.util.Map;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpStatusCode;
```
HttpStatus is imported twice. Remove the import statement for org.springframework.http.HttpStatusCode as it is not required.

slade13 reviewed (changes requested):
DTO class validation was added to `CreateBookRequestDto` but there are also 2 classes which are missing validation: `BookDto` and `BookSearchParameters`

jszymanski97 committed:
CoverImage interface and Validator created
validations added for BookDto and BookSearchParameters
extra validations added to CreateBookRequestDto

slade13 reviewed (approved):
Great job!
